File: wellness/wellness.Model/Reservation/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace wellness.Model.Reservation
{
    public class Reservation
    {
        public int Id { get; set; }

        public string FirstName { get; set; } = null!;

        public string LastName { get; set; } = null!;

        public string Date { get; set; } = null!;

        public string Time { get; set; } = null!;

        public string Phone { get; set; }= null!;

        public bool? Status { get; set; }

        public string Treatment { get; set; } = null!;
        public int TreatmentId { get; set; }
    }
}

File: wellness/UI/desktop/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int id;
  final String firstName;
  final String lastName;
  final String date;
  final String time;
  final String phone;
  final bool? status;
  final String treatment;

  Reservation(this.id, this.firstName, this.date, this.time, this.status,
      this.treatment, this.lastName, this.phone);

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
